# 产生目录

目录中英文映射：

In [1]:
folders = Dict(
  "01-julia-overview" => "01. **Julia 快速概览**",
  "02-julia-basics" => "02. **Julia 基础**",
  "03-julia-syntax" => "03. **Julia 语法进阶**",
  "04-julia-oop" => "04. **Julia 面向对象编程**",
  "05-julia-on-ml" => "05. **Julia 机器学习编程**"
)

Dict{String,String} with 5 entries:
  "02-julia-basics"   => "02. **Julia 基础**"
  "01-julia-overview" => "01. **Julia 快速概览**"
  "04-julia-oop"      => "04. **Julia 面向对象编程**"
  "03-julia-syntax"   => "03. **Julia 语法进阶**"
  "05-julia-on-ml"    => "05. **Julia 机器学习编程**"

In [2]:
using Pkg
Pkg.add("JSON")

  Updating registry at `~/.julia/registries/General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
 Resolving package versions...
  Updating `~/.julia/environments/v1.1/Project.toml`
 [no changes]
  Updating `~/.julia/environments/v1.1/Manifest.toml`
 [no changes]


产生目录文件：

In [3]:
import JSON

function get_title(ipynb_file, default)
    content = open(f -> read(f, String), ipynb_file)
    obj = JSON.parse(content)
    cell = obj["cells"][1]
    first_line = cell["source"][1]
    if first_line == nothing
        default
    else
        index = match(r"^[\d.]+", default).match
        title = replace(chomp(first_line), r"\#\s*" => "")
        "$index. **$title**"
    end
end

get_title (generic function with 1 method)

In [4]:
open("table-of-contents.md", "w") do io
    for folder in sort(collect(keys(folders)))
        folder_name = folders[folder]
        write(io, "- [$folder_name]($folder)\n")
        for f in filter(x -> endswith(x, ".ipynb"), readdir(folder))
            current_file = "$folder/$f"
            title = strip(get_title(current_file, f))
            write(io, "\t- [$title]($current_file)\n")
        end
    end
end

content = open(f -> read(f, String), "table-of-contents.md")
println(content)

- [01. **Julia 快速概览**](01-julia-overview)
	- [01.01. **Ruby 简介**](01-julia-overview/01.01-julia-overview.ipynb)
	- [01.02. **20分钟体验 Ruby**](01-julia-overview/01.02-julia-in-20-minutes.ipynb)
	- [01.03. **从其它语言到 Ruby**](01-julia-overview/01.03-julia-from-other-languages.ipynb)
- [02. **Julia 基础**](02-julia-basics)
	- [02.01. **Ruby 数据类型**](02-julia-basics/02.01-types.ipynb)
	- [02.02. **数字**](02-julia-basics/02.02-numbers.ipynb)
	- [02.03. **字符串**](02-julia-basics/02.03-strings.ipynb)
	- [02.04. **符号**](02-julia-basics/02.04-symbols.ipynb)
	- [02.05. **正则表达式**](02-julia-basics/02.05-regular-expression.ipynb)
	- [02.06. **列表**](02-julia-basics/02.06-arrays.ipynb)
	- [02.07. **Hash**](02-julia-basics/02.07-hashes.ipynb)
	- [02.08. **范围 (Range)**](02-julia-basics/02.08-ranges.ipynb)
- [03. **Julia 语法进阶**](03-julia-syntax)
	- [03.01. **Ruby 赋值机制**](03-julia-syntax/03.01-assignment-statements.ipynb)
	- [03.02. **判断语句**](03-julia-syntax/03.02-conditional-statements.ipynb)
	- [03.03. **循环**](0

- [01. **Julia 快速概览**](01-julia-overview)
	- [01.01. **Ruby 简介**](01-julia-overview/01.01-julia-overview.ipynb)
	- [01.02. **20分钟体验 Ruby**](01-julia-overview/01.02-julia-in-20-minutes.ipynb)
	- [01.03. **从其它语言到 Ruby**](01-julia-overview/01.03-julia-from-other-languages.ipynb)
- [02. **Julia 基础**](02-julia-basics)
	- [02.01. **Ruby 数据类型**](02-julia-basics/02.01-types.ipynb)
	- [02.02. **数字**](02-julia-basics/02.02-numbers.ipynb)
	- [02.03. **字符串**](02-julia-basics/02.03-strings.ipynb)
	- [02.04. **符号**](02-julia-basics/02.04-symbols.ipynb)
	- [02.05. **正则表达式**](02-julia-basics/02.05-regular-expression.ipynb)
	- [02.06. **列表**](02-julia-basics/02.06-arrays.ipynb)
	- [02.07. **Hash**](02-julia-basics/02.07-hashes.ipynb)
	- [02.08. **范围 (Range)**](02-julia-basics/02.08-ranges.ipynb)
- [03. **Julia 语法进阶**](03-julia-syntax)
	- [03.01. **Ruby 赋值机制**](03-julia-syntax/03.01-assignment-statements.ipynb)
	- [03.02. **判断语句**](03-julia-syntax/03.02-conditional-statements.ipynb)
	- [03.03. **循环**](03-julia-syntax/03.03-loop-statements.ipynb)
	- [03.04. **迭代器**](03-julia-syntax/03.04-iterators.ipynb)
	- [03.05. **函数**](03-julia-syntax/03.05-functions.ipynb)
	- [03.06. **代码块**](03-julia-syntax/03.06-blocks.ipynb)
	- [03.07. **变量作用域**](03-julia-syntax/03.07-scope.ipynb)
	- [03.09. **异常**](03-julia-syntax/03.09-exceptions.ipynb)
	- [03.10. **Ruby 元编程技巧**](03-julia-syntax/03.10-julia-meta-programming.ipynb)
	- [03.11. **Ruby 代码风格**](03-julia-syntax/03.11-julia-style-guide.ipynb)
- [04. **Julia 面向对象编程**](04-julia-oop)
- [05. **Julia 机器学习编程**](05-julia-on-ml)